tests/testthat/test-io-parquet-functions.R

```diff
@@ -138,132 +138,140 @@ test_that("scan_parquet can include file path", {
-#       "gear=3.0/cyl=8.0/00000000.parquet", "gear=4.0/cyl=4.0/00000000.parquet",
-#       "gear=4.0/cyl=6.0/00000000.parquet", "gear=5.0/cyl=4.0/00000000.parquet",
-#       "gear=5.0/cyl=6.0/00000000.parquet", "gear=5.0/cyl=8.0/00000000.parquet"
-#     )
-#   )
-
-#   # works fine with integers
-#   temp_dir <- withr::local_tempdir()
-#   dat2 <- dat$with_columns(pl$col("gear")$cast(pl$Int32), pl$col("cyl")$cast(pl$Int32))
-#   dat2$write_parquet(temp_dir, partition_by = c("gear", "cyl"))
-#   expect_equal(
-#     list.files(temp_dir, recursive = TRUE),
-#     c(
-#       "gear=3/cyl=4/00000000.parquet", "gear=3/cyl=6/00000000.parquet",
-#       "gear=3/cyl=8/00000000.parquet", "gear=4/cyl=4/00000000.parquet",
-#       "gear=4/cyl=6/00000000.parquet", "gear=5/cyl=4/00000000.parquet",
-#       "gear=5/cyl=6/00000000.parquet", "gear=5/cyl=8/00000000.parquet"
-#     )
-#   )
-
-#   # check inputs
-#   expect_grepl_error(
-#     dat$write_parquet(temp_dir, partition_by = "foo"),
-#     r"("foo" not found)"
-#   )
-#   expect_grepl_error(
-#     dat$write_parquet(temp_dir, partition_by = ""),
-#     r"("" not found)"
-#   )
-#   expect_grepl_error(dat$write_parquet(temp_dir, partition_by = 1))
-# })
-
-# test_that("polars and arrow create the same hive partition", {
-#   skip_if_not_installed("arrow")
-#
-#   # arrow
-#   temp_dir_arrow <- withr::local_tempdir()
-#   dat <- mtcars
-#   on.exit(unlink(temp_dir_arrow))
-#   arrow::write_dataset(
-#     dat,
-#     temp_dir_arrow,
-#     partitioning = c("cyl", "gear"),
-#     format = "parquet",
-#     hive_style = TRUE
-#   )
-
-#   # polars
-#   temp_dir_polars <- withr::local_tempdir()
-#   dat2 <- as_polars_df(mtcars)$
-#     with_columns(pl$col("gear")$cast(pl$Int32), pl$col("cyl")$cast(pl$Int32))
-#   on.exit(unlink(temp_dir_polars))
-#   dat2$write_parquet(temp_dir_polars, partition_by = c("cyl", "gear"))
-
-#   # check dirnames because filenames are different between the two
-#   expect_equal(
-#     dirname(list.files(temp_dir_arrow, recursive = TRUE)),
-#     dirname(list.files(temp_dir_polars, recursive = TRUE))
-#   )
-# })
+test_that("write_parquet works", {
+  tmpf <- withr::local_tempfile()
+  on.exit(unlink(tmpf))
+  df_exp <- as_polars_df(mtcars)
+  df_exp$write_parquet(tmpf)
+
+  expect_equal(
+    pl$read_parquet(tmpf),
+    mtcars,
+    ignore_attr = TRUE
+  )
+})
+
+test_that("throw error if invalid compression is passed", {
+  tmpf <- withr::local_tempfile()
+  on.exit(unlink(tmpf))
+  df_exp <- as_polars_df(mtcars)
+  expect_error(
+    df_exp$write_parquet(tmpf, compression = "invalid"),
+    "must be one of"
+  )
+})
+
+test_that("write_parquet returns the input data", {
+  dat <- as_polars_df(mtcars)
+  tmpf <- withr::local_tempfile()
+  x <- dat$write_parquet(tmpf)
+  expect_equal(x, dat)
+})
+
+test_that("write_parquet: argument 'statistics'", {
+  dat <- as_polars_df(mtcars)
+  tmpf <- withr::local_tempfile()
+  on.exit(unlink(tmpf))
```

Comment:
IIUC, the file created by ` withr::local_tempfile` will be removed automatically, so we don't need this.